### DISCUSSION TITLE
Multiple roles per user

### DISCUSSION THREAD

**Discussion by HimoriTempest** [directus/directus#20593] [Feature Requests]
### Summary

Ability to assign multiple roles per user and combine their permissions

### Motivation

Why? Being able to combine multiple roles instead of creating a new role.
Use Case: Being able to separate permissions into groups which can be combined without having to create a new role for that case

### Detailed Design

- Role A => Permission to read Table A
- Role B => Permission to read Table B
- Role C => Permission to edit table B
- Role D => Permission to read Table C

User Nils got roles A & B an can therefore read both tables 
User Alex got roles D & B & C which results in them being able to Read C / B as well as Edit B

### Must Have:
- Multiselect for Roles on User creation / Profile page

### Could Have:
- Permission inheritance between parent and child roles

### Drawbacks

It would make figuring out permissions per user more complex based on the number of roles they have
It would break:
- Kanban layout for roles as multiple users can have multiple roles
- anything that relies on the role property in a user to be a string instead of an array

### Alternatives

Possible alternative: Primary role + Secondary roles

### Adoption Strategy

This is a breaking change, reading a users roles would result in a array instead of a role id string
A possible migration could be: Keeping the current property as primary role and adding a secondary roles property to deal with additional roles

**Comment by rijkvanzanten** [comment]
I believe this is a duplicate of https://github.com/directus/directus/discussions/4959

**Reply by HimoriTempest** [reply]
Makes sense especially the approach of having policies. Yeah this can be closed
